src/realm/sync/instruction_applier.cpp

```diff
@@ -1493,13 +1490,20 @@ InstructionApplier::PathResolver::Status InstructionApplier::PathResolver::resol
                 ++m_it_begin;
                 return resolve_dictionary_element(dict, *pkey);
             }
+            else {
+                // TODO: this needs to be fixed..
+                return InstructionApplier::PathResolver::Status::Pending;
```

Comment:
this is not OK, it needs a proper fixing.. but the checks as we discussed with @danieltabacaru  are wrong.